File: algorithms/algorithms-cplusplus/leetcode/Question_0060_Permutation_Sequence.cc
#include <string>
#include <vector>

using namespace std;

class Solution {
 public:
  string getPermutation(int n, int k) {
    vector<int> nums(n);
    int cnt = 1;
    for (int i = 0; i < n; i++) {
      nums[i] = i + 1;
      cnt *= i + 1;
    }
    k--;
    string res = "";
    for (int i = 0; i < n; i++) {
      cnt /= n - i;
      int select = k / cnt;
      res += '0' + nums[select];
      for (int j = select; j < n - i - 1; j++) {
        nums[j] = nums[j + 1];
      }
      k = k % cnt;
    }
    return res;
  }
};

File: algorithms/algorithms-python/leetcode/Question_0060_Permutation_Sequence.py
#!/usr/bin/python
# coding: utf-8

class Solution(object):
    def getPermutation(self, n, k):
        """
        :type n: int
        :type k: int
        :rtype: str
        """
        nums = [0] * n
        pCount = 1
        for i in range(n):
            nums[i] = i + 1
            pCount *= i + 1

        k -= 1
        res = []
        for i in range(n):
            pCount /= n - i
            select = k / pCount
            res.append(str(nums[select]))
            nums.remove(nums[select])
            k %= pCount

        return "".join(res)
